activerecord/lib/active_record/scoping/named.rb

```diff
@@ -30,13 +30,8 @@ def all
         end
 
         def default_scoped # :nodoc:
-          scope = build_default_scope
-
-          if scope
-            relation.spawn.merge!(scope)
-          else
-            relation
-          end
+          scope = relation
+          build_default_scope(scope) || scope
```

Comment:
Also, is this safe? It sounds like it could change which relation the default_scope block sees, or something.